### PR TITLE
Translation pt br - Thinking in Components

### PR DESCRIPTION
**Pull request by alvarocamillont**

### CODE REVIEW

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+<script async custom-element="amp-youtube" src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>
+```
+
+[tip type="note"]
+**Nota**: AMP também inclui suporte para outros players de vídeo. Confira "[Integrando Vídeos em AMP e Visão Geral](https://ampbyexample.com/advanced/integrating_videos_in_amp_an_overview/)" em AMP por exemplos.[/tip]
+[/filter]
+[filter level="advanced"]
+Há apenas um erro de validação restante em nosso site, e vamos corrigi-lo usando um **componente** `<amp-img>` em vez de uma tag `<img>`.
+
+## Oque são Web Components?
```
Comment (fellyph) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:144]: space for 'O que'

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+Os componentes são blocos de construção para a web. Eles combinam estrutura (HTML), estilo (CSS) e comportamento (JavaScript) com uma interface simples. Os componentes têm:
+
+- Um nome (por exemplo, `<amp-img>`) usado como o nome da tag para identificar o componente.
+
+- Atributos personalizados que alteram o comportamento, estilo ou conteúdo de um componente (como `width`,` height`, `src` e`attribution`).
+
+- Eventos que podem capturar entradas do usuário para o componente (atributo ʻon`).
```
Comment (fellyph) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:152]: apostrophe and backticks in the word on probably should be backticks and backticks or apostrophe and apostrophe.

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+**Dica**: Tente selecionar diferentes dispositivos móveis no menu suspenso (veja a captura de tela abaixo) para ver como a imagem se adapta a diferentes tamanhos de tela. É uma boa prática testar seu site em tamanhos de tela diferentes. Navegadores em dispositivos móveis reais podem se comportar de maneira diferente, então, quando possível, teste sua página da web em dispositivos reais também.
+
+{{ image('/static/img/courses/beginner/image4.png', 297, 380,  align='center half', caption='Lista suspensa de dispositivos no Chrome') }}
+[/tip]
+
+## Exercício 4: Incorporação de Vídeo
+
+A seguir, vamos incorporar um vídeo do YouTube em nosso documento. Nossa equipe de marketing da Chico’s Bikes lançou [este](https://www.youtube.com/watch?v=BlpMQ7fMCzA) vídeo de uma de nossas bicicletas queijos sendo construídas.
+
+Use a documentação do [`<amp-youtube>`](../../../documentation/components/reference/amp-youtube.md) para incorporar este vídeo do YouTube no componente `<amp-img>` com as seguintes configurações:
```
Comment (fellyph) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:107]: is possible to add a locale path in those links? 

Such as:
```
[apêndice]({{g.doc('/content/amp-dev/documentation/courses/beginning-course/appendix.md', locale=doc.locale).url.path}}).
```

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+Se você deseja se tornar um desenvolvedor de AMP de sucesso, aprender a usar o sistema de layout é fundamental. Todos os layouts fornecidos pelo AMP podem ser implementados usando CSS simples, mas geralmente podem ser complicados ou ter casos extremos complicados que exigem conhecimento profundo para serem solucionados. O AMP simplifica o processo e expõe muitas dessas opções de layout para serem usadas em qualquer elemento da página AMP. Confira o
+[documentação oficial](../../../documentation/guides-and-tutorials/learn/amp-html-layout/index.md) para obter mais informações sobre o sistema de layout.
```
Comment (fellyph) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:95]: same as the following comment related to the locale path

**Review by fellyph — COMMENTED** [review]
More structural comments related to the PR

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md (hunk 2 above)
Comment (alvarocamillont) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:152]: Done

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md (hunk 3 above)
Comment (alvarocamillont) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:107]: Done!

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md (hunk 4 above)
Comment (alvarocamillont) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:95]: Done!

**Comment by alvarocamillont** [comment]
@fellyph I think that's all. :)

**Review by fellyph — APPROVED** [review]
All Good for me, thanks again @alvarocamillont

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+Mas essa é a beleza do AMP. Com AMP, não precisamos nos preocupar com todos esses detalhes! Os autores da biblioteca AMP nos forneceram blocos de construção que nos fornecem esses recursos e ajudam a cuidar de qualidades como desempenho, acessibilidade e segurança. Esses blocos são chamados de **componentes** e são a chave para criar sites de sucesso com AMP.
+
+## O que são Web Components?
+
+Os componentes são blocos de construção para a web. Eles representam a combinação de estrutura (HTML), estilo (CSS) e comportamento (JavaScript) com uma interface que facilita o uso em seu site e o compartilhamento com outras pessoas. Os componentes têm:
+
+- Um nome (por exemplo, `<amp-img>`) usado como o nome da tag para identificar o componente.
+
+- Atributos personalizados que alteram o comportamento, estilo ou conteúdo de um componente (como `width`,` height`, `src` e`attribution`).
```
Comment (morsssss) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:24]: can you insert a space before "attribution"?

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+### Solução
+
+A parte da página que contém a imagem deve ter a seguinte aparência:
+
+[sourcecode:html]
+{% raw %}<amp-img src="https://cdn.glitch.com/d7f46a57-0ca4-4cca-ab0f-69068dec6631%2Fricotta-racer.jpg?1540228217746" width=”640” height="480"></amp-img>
+{% endraw %}[/sourcecode]
+
+[tip type="note"]
+**Nota**:: Verifique a extensão do Validador de AMP. Se você concluiu este exercício com sucesso, o ícone da extensão deve estar verde. Se sim, parabéns! Sua página agora é AMP válida!
```
Comment (morsssss) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:68]: got an extra colon here...

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+
+Por exemplo, no caso de `<amp-img>`, AMP exige que especifiquemos as dimensões da imagem. O AMP precisa entender o layout da página antes do carregamento dos recursos (como imagens). Isso melhora a experiência do usuário quando sua página está carregando, mas antes que os ativos de imagem tenham sido baixados. Quando as imagens são baixadas, elas podem ser inseridas na página sem que nenhum conteúdo existente na página se mova. Isso dá ao runtime do AMP margem para decidir quando carregar ativos de imagem com base nos recursos do dispositivo do usuário e na conexão com a Internet.
+
+[tip type="note"]
+**Note**: Se você quiser saber mais sobre as otimizações automáticas de AMP, leia sobre o lazy-loading em AMP no [apêndice]({{g.doc('/content/amp-dev/documentation/courses/beginning-course/appendix.md', locale=doc.locale).url.path}}).
+[/tip]
+
+Para usar o componente e resolver o erro de validação `<amp-img>` anterior, substitua a tag img existente em nossa página pelo equivalente de AMP.
+Dica: em vez de `<img…>`, escreva `<amp-img…>` e atribua dimensões fixas à imagem. Dê à imagem uma `largura` de` 640` e uma `altura` de` 480`.
```
Comment (morsssss) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:55]: For some reason, when you place these sentences (starting with "Dica:") on a new line, it confuses the parser. I don't know why that is... but if you remove the newline before "Dica:", it does fix the problem!

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md [hunk 8]
```diff
@@ -0,0 +1,265 @@
+---
+$title: Pensando em componentes
+$order: 4
+leveled: true
+---
+
+[filter level="beginner"]
+## Adicionando recursos ao nosso site
+
+Até agora, convertemos nosso site HTML básico em um site AMP básico. Há apenas um erro de validação restante em nosso site, e vamos corrigi-lo usando um componente `<amp-img>` em vez de uma tag `<img>`. Conforme corrigimos esse último erro de validação, aprenderemos o que são os componentes de AMP, por que algumas tags HTML foram substituídas ou não permitidas em AMP e como adicionar componentes ao nosso site.
```
Comment (morsssss) [inline pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:10]: I leave this question to you and to @fellyph ... I assume that it's ok to use the Portuguese "componentes" throughout this article, even though we also use the English phrase "Web Components"?

**Review by morsssss — CHANGES_REQUESTED** [review]

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md (hunk 5 above)
Comment (alvarocamillont) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:24]: Done

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md (hunk 6 above)
Comment (alvarocamillont) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:68]: Done!

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md (hunk 7 above)
Comment (alvarocamillont) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:55]: Done!

File: pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md (hunk 8 above)
Comment (alvarocamillont) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:10]: When the concept was about web components specifically I leave it in English, when the concept was generic as a "component" I translated it into Portuguese. Do you agree @fellyph?

Comment (morsssss) [reply pages/content/amp-dev/documentation/courses/beginning-course/thinking-in-components@pt_BR.md:10]: That makes sense to me! I've already delayed this enough, so I want to go ahead and approve. If Fellyph thinks this should be changed, well, that's still possible!

**Review by morsssss — APPROVED** [review]

**Pull request merged**